pkg/controller/ingress-shim/sync.go

```diff
@@ -180,6 +181,22 @@ func (c *Controller) buildCertificates(ing *extv1beta1.Ingress, issuer v1alpha1.
 		if existingCrt != nil {
 			klog.Infof("Certificate %q for ingress %q already exists", tls.SecretName, ing.Name)
 
+			ownerRefs := existingCrt.GetOwnerReferences()
+			if len(ownerRefs) == 0 {
+				klog.Infof("Certificate %q has no owners and cannot be updated for ingress %q", tls.SecretName, ing.Name)
+				continue
+			}
+
+			notOwned := false
+			for _, ownerRef := range ownerRefs {
+				notOwned = notOwned || !reflect.DeepEqual(ingOwnerRef, ownerRef)
```

Comment:
I'll defer a proper review to Louis or myself later on when it isn't 3.30am here, but there is an IsControlledBy function we can use instead of reflect.DeepEqual: https://godoc.org/k8s.io/apimachinery/pkg/apis/meta/v1#IsControlledBy

Resources *can* technically have multiple owner references, although only one of those may have the contolled-by field set. This could trip up the current reflect.DeepEqual codepath in some very rare caaes!